parse/pdf.js

```diff
@@ -0,0 +1,424 @@
+  
+  for(let i in versions) {
+    versions[i] = versions[i].replace(`${path}version-`,"")
+  }
+
+  versions.push('next');
+
+  return versions;
+}
+
+/**
+ * Request a path to output to, creating it if necessary
+ * @param {string} req (Optional) The path to the requested output directory
+ * @param {string} def (Optional) The default path to output to
+ * @return {string} The absolute path to the output directory without the trailing slash
+ */
+function setOutputDir(req='',def=outputPath) {
+  if(req) {
+    let userOutputPath = paths.resolve(req);
+  
+    fs.access(userOutputPath, fs.constants.W_OK, function(err) {
+      if(err){
+        req = def;
+        process.emitWarning(`Could not write to ${userOutputPath}, instead writing to ${def}`);
+      }
+  
+      req = userOutputPath;
+    });
+  }
+  else {
+    req = def
+  }
+  
+  // create the output folder if it doesn't exist
+  if(!fs.existsSync(req)) {
+    fs.mkdirSync(req);
+  }
+
+  return req;
+}
+
+/**
+ * Converts `filename` to a MarkDown title anchor link.
+ * @param {string} filename Name of the file, e.g. `cues/creating-a-cue.md`
+ * @return {string} A MarkDown title anchor link, e.g. `#cues-creating-a-cue.md`
+ */
+function filenameToTitleLink(filename) {
+  return "#" + filename.replace("/","-");
+}
+
+/**
+ * Replaces the YAML block in the file with a heading
+ * @param {string} filename Name of the file, e.g. `cues/creating-a-cue.md`
+ * @param {string} content Contents of the .md file with YAML block in
+ * @return {string} The content with the YAML block replaced
+ */
+function replaceYaml(filename,content) {
+  // matches Yaml block with title
+  return content.replace(/^---(?:[\n]|.)*title: *([\w ]*)(?:[\n]|.(?!--))*---/mgi,function (match,title) {
+    titleLink = filenameToTitleLink(filename);
+    if (filename.match("/")) {
+      // sub page, e.g.:
+      // # Cues {#cues.md}
+      return `# ${title} {${titleLink}}`;
+    }
+    else {
+      // heading page, e.g.:
+      // # {#cues.md}
+      // \part{Cues}
+      return `# {${titleLink}}\n\\part{${title}}`;
+    }
+  });
+}
+
+/**
+ * Replaces links to local MarkDown files with links to the title
+ * @param {string} filename Name of the file, e.g. `cues/creating-a-cue.md`
+ * @param {string} content Contents of the .md file with the links in
+ * @param {string} docsPath The path to the docs folder, e.g. `../docs/`
+ * @return {string} The content with the images fixed
+ */
+function replaceLinks(filename,content,docsPath) {
+  // matches all links which are to local .md files
+  return content.replace(/(?<!!)\[([^\[]*\n*)\]\((?!https?:\/\/)(?!\/\/)(?!#)([a-zA-Z0-9-\.\/]*\.md)([^)]*)\)/mgi, function (match,text,link,anchor) {
+    if(anchor) {
+      // if it's got an anchor link to a title just go to that, e.g.
+      // change [text](link.md#title)
+      // to     [text](#title)
+      return `[${text}](${anchor})`
```

Comment:
Is there a risk here with duplicate headings? For example if there were multiple headings called "Create" would this cause an issue?

Is this for the whole document or just this part?

Reply:
Yes you're right, it was to save adding IDs to every heading, e.g.

````markdown
# Create {cues-creating-a-cue-md-create}
````

*Alternatively I could just make it link to page like the bit below does*

But I've also just looked at this and realised that this won't warn if the file it's linking to 404s... so could add that?

Reply:
Resolved that last bit in e808fa4